server/src/main/java/org/elasticsearch/cluster/metadata/IndexTemplateMetadata.java

```diff
@@ -378,7 +382,9 @@ private static void toInnerXContent(IndexTemplateMetadata indexTemplateMetadata,
             indexTemplateMetadata.settings().toXContent(builder, params);
             builder.endObject();
 
-            includeTypeName &= (params.paramAsBoolean("reduce_mappings", false) == false);
+            if(builder.getRestApiVersion().matches(onOrAfter(V_8))) {
+                includeTypeName &= (params.paramAsBoolean("reduce_mappings", false) == false);
+            }
```

Comment:
@romseygeek do you think there need to be a special handing for this in V7 mode?
The logic for 7.x was more complex

Reply:
I think this `reduce_mappings` parameter is only set internally, so I don't think we want to put a Rest API version guard on it.  It's for things like template upgrades or building internal indexes from templates.